<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>921</width>
    <height>640</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Scene Graph</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="MyGL" name="mygl">
    <property name="geometry">
     <rect>
      <x>11</x>
      <y>11</y>
      <width>618</width>
      <height>592</height>
     </rect>
    </property>
   </widget>
   <widget class="QTreeWidget" name="treeWidget">
    <property name="geometry">
     <rect>
      <x>640</x>
      <y>10</y>
      <width>256</width>
      <height>381</height>
     </rect>
    </property>
    <column>
     <property name="text">
      <string notr="true">1</string>
     </property>
    </column>
   </widget>
   <widget class="QDoubleSpinBox" name="txSpinBox">
    <property name="geometry">
     <rect>
      <x>650</x>
      <y>430</y>
      <width>62</width>
      <height>22</height>
     </rect>
    </property>
    <property name="minimum">
     <double>-98.000000000000000</double>
    </property>
    <property name="singleStep">
     <double>0.500000000000000</double>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="tySpinBox">
    <property name="geometry">
     <rect>
      <x>720</x>
      <y>430</y>
      <width>62</width>
      <height>22</height>
     </rect>
    </property>
    <property name="minimum">
     <double>-98.000000000000000</double>
    </property>
    <property name="singleStep">
     <double>0.500000000000000</double>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="rSpinBox">
    <property name="geometry">
     <rect>
      <x>650</x>
      <y>480</y>
      <width>62</width>
      <height>22</height>
     </rect>
    </property>
    <property name="minimum">
     <double>-180.000000000000000</double>
    </property>
    <property name="maximum">
     <double>180.000000000000000</double>
    </property>
    <property name="singleStep">
     <double>0.500000000000000</double>
    </property>
   </widget>
   <widget class="QLabel" name="txLabel">
    <property name="geometry">
     <rect>
      <x>650</x>
      <y>410</y>
      <width>61</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Translate X</string>
    </property>
   </widget>
   <widget class="QLabel" name="tyLabel">
    <property name="geometry">
     <rect>
      <x>720</x>
      <y>410</y>
      <width>61</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Translate Y</string>
    </property>
   </widget>
   <widget class="QLabel" name="rLabel">
    <property name="geometry">
     <rect>
      <x>650</x>
      <y>460</y>
      <width>61</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Rotate</string>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="sxSpinBox">
    <property name="geometry">
     <rect>
      <x>650</x>
      <y>530</y>
      <width>62</width>
      <height>22</height>
     </rect>
    </property>
    <property name="singleStep">
     <double>0.500000000000000</double>
    </property>
   </widget>
   <widget class="QLabel" name="sxLabel">
    <property name="geometry">
     <rect>
      <x>650</x>
      <y>510</y>
      <width>61</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Scale X</string>
    </property>
   </widget>
   <widget class="QDoubleSpinBox" name="sySpinBox">
    <property name="geometry">
     <rect>
      <x>720</x>
      <y>530</y>
      <width>62</width>
      <height>22</height>
     </rect>
    </property>
    <property name="singleStep">
     <double>0.500000000000000</double>
    </property>
   </widget>
   <widget class="QLabel" name="syLabel">
    <property name="geometry">
     <rect>
      <x>720</x>
      <y>510</y>
      <width>61</width>
      <height>16</height>
     </rect>
    </property>
    <property name="text">
     <string>Scale Y</string>
    </property>
   </widget>
   <widget class="QPushButton" name="geomSetButton">
    <property name="geometry">
     <rect>
      <x>800</x>
      <y>580</y>
      <width>111</width>
      <height>21</height>
     </rect>
    </property>
    <property name="cursor">
     <cursorShape>PointingHandCursor</cursorShape>
    </property>
    <property name="text">
     <string>Set Geometry</string>
    </property>
   </widget>
   <widget class="QPushButton" name="tNodeAddButton">
    <property name="geometry">
     <rect>
      <x>800</x>
      <y>430</y>
      <width>111</width>
      <height>21</height>
     </rect>
    </property>
    <property name="cursor">
     <cursorShape>PointingHandCursor</cursorShape>
    </property>
    <property name="text">
     <string>Add Translate Node</string>
    </property>
   </widget>
   <widget class="QPushButton" name="rNodeAddButton">
    <property name="geometry">
     <rect>
      <x>800</x>
      <y>480</y>
      <width>111</width>
      <height>21</height>
     </rect>
    </property>
    <property name="cursor">
     <cursorShape>PointingHandCursor</cursorShape>
    </property>
    <property name="text">
     <string>Add Rotate Node</string>
    </property>
   </widget>
   <widget class="QPushButton" name="sNodeAddButton">
    <property name="geometry">
     <rect>
      <x>800</x>
      <y>530</y>
      <width>111</width>
      <height>21</height>
     </rect>
    </property>
    <property name="cursor">
     <cursorShape>PointingHandCursor</cursorShape>
    </property>
    <property name="text">
     <string>Add Scale Node</string>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>921</width>
     <height>25</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionQuit"/>
   </widget>
   <addaction name="menuFile"/>
  </widget>
  <action name="actionQuit">
   <property name="text">
    <string>Quit</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Q</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>MyGL</class>
   <extends>QOpenGLWidget</extends>
   <header>mygl.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
